plugin/backend.go

```diff
@@ -104,6 +108,28 @@ func (b *GcpAuthBackend) IAM(s logical.Storage) (*iam.Service, error) {
 	return b.iamClient, nil
 }
 
+func (b *GcpAuthBackend) GCE(s logical.Storage) (*compute.Service, error) {
+	b.clientMutex.RLock()
```

Comment:
Is there a chance that you'll have the same mount being used for GCE and IAM? Or is it strictly one or the other?

Reply:
OK, I see the answer is yes. It might make things more complicated, might not, but may be worth splitting out the mutex so it's one per type. But I doubt there will be contention issues if not, so feel free not to!

Reply:
I'll keep it as is and if someone actually runs into an issue with it, we can change it :)